src/service/extensions-impl.js

```diff
@@ -250,19 +243,39 @@ export class Extensions {
    * Returns the promise that will be resolved when the extension has been
    * loaded. If necessary, adds the extension script to the page.
    * @param {string} extensionId
-   * @param {boolean=} stubElement
    * @return {!Promise<!ExtensionDef>}
    */
-  loadExtension(extensionId, stubElement = true) {
+  preloadExtension(extensionId) {
     if (extensionId == 'amp-embed') {
       extensionId = 'amp-ad';
     }
     const holder = this.getExtensionHolder_(extensionId);
-    this.insertExtensionScriptIfNeeded_(extensionId, holder, stubElement);
+    this.insertExtensionScriptIfNeeded_(extensionId, holder);
     return this.waitFor_(holder);
   }
 
   /**
+   * Returns the promise that will be resolved when the extension has been
+   * loaded. If necessary, adds the extension script to the page.
+   * @param {!./ampdoc-impl.AmpDoc} ampdoc
+   * @param {string} extensionId
+   * @return {!Promise<!ExtensionDef>}
+   */
+  installExtensionForDoc(ampdoc, extensionId) {
+    const rootNode = ampdoc.getRootNode();
+    let extLoaders = rootNode[LOADER_PROP];
+    if (!extLoaders) {
+      extLoaders = rootNode[LOADER_PROP] = map();
+    }
+    if (extLoaders[extensionId]) {
+      return extLoaders[extensionId];
+    }
+    stubElementIfNotKnown(ampdoc.win, extensionId);
+    return extLoaders[extensionId] = this.preloadExtension(extensionId)
+        .then(() => this.installExtensionInDoc_(ampdoc, extensionId));
```

Comment:
Yes! The `installExtensionInDoc_` replaced `runExtensionFactoriesInDoc_`. I figured "install" name is better, since it's now became explicitly installation.